packages/nyc-test-coverage/src/index.ts

```diff
@@ -0,0 +1,34 @@
+import { InjectedSinks } from '@temporalio/worker';
+import { CoverageSinks } from './sinks';
+import libCoverage from 'istanbul-lib-coverage';
+
+export function createCoverageSinks(): { sinks: InjectedSinks<CoverageSinks>; coverageMap: libCoverage.CoverageMap } {
+  const coverageMap = libCoverage.createCoverageMap();
+
+  return {
+    coverageMap,
+    sinks: {
+      coverage: {
+        merge: {
+          fn(_workflowInfo: any, testCoverage: libCoverage.CoverageMap) {
+            coverageMap.merge(testCoverage);
+          },
+          callDuringReplay: false,
+        },
+      },
+    },
+  };
+}
+
+export function mergeWorkflowCoverage(coverageMap: libCoverage.CoverageMap): void {
```

Comment:
At this point I'd just create a class (or instance with defined interface if you prefer) with a `sinks` getter and a `mergeWorkflowCoverage` method.

These methods require docs too.

I'd go with:

```ts
class WorkflowCoverage {
  get sinks(): InjectedSinks<CoverageSinks> { ... }
  mergeIntoGlobalCoverage(): void {}
}
```